app/lib/service/openid/jwt.dart

```diff
@@ -123,7 +114,179 @@ Map<String, dynamic> _decodePart(String part) {
   return _jsonUtf8Base64.decode(base64.normalize(part)) as Map<String, dynamic>;
 }
 
-DateTime? _tryParseFromSeconds(int? value) {
+DateTime? _parseSecondsIfNotNull(int? value) {
   if (value == null) return null;
-  return DateTime.fromMillisecondsSinceEpoch(value * 1000);
+  return DateTime.fromMillisecondsSinceEpoch(value * 1000, isUtc: true);
+}
+
+/// The parsed JWT header.
+class JwtHeader extends UnmodifiableMapView<String, dynamic> {
+  /// algorithm
+  final String? alg;
+
+  /// type
+  final String? typ;
+
+  /// key identifier
+  final String? kid;
+
+  JwtHeader._(super.map)
+      : alg = map['alg'] as String?,
+        typ = map['typ'] as String?,
+        kid = map['kid'] as String?;
+
+  factory JwtHeader(Map<String, dynamic> values) {
+    try {
+      return JwtHeader._(values);
+    } on FormatException {
+      rethrow;
+    } catch (_) {
+      throw FormatException('Unexpected value in JWT header.');
+    }
+  }
+}
+
+/// The parsed JWT payload.
+class JwtPayload extends UnmodifiableMapView<String, dynamic> {
```

Comment:
For the record `JwtPayload` and `JwtHeader` are pretty nice ways of structuring things.